libraries/AC_AutoTune/AC_AutoTune_FreqResp.h

```diff
@@ -6,13 +6,14 @@
 
 #include <AP_Math/AP_Math.h>
 
-#define AUTOTUNE_DWELL_CYCLES                6
-
 class AC_AutoTune_FreqResp {
 public:
     // Constructor
     AC_AutoTune_FreqResp()
 {
+    // ring buffers sized to for more cycles than are needed.  Most cycles needed are 6.
+    meas_peak_info_buffer = NEW_NOTHROW ObjectBuffer<peak_info>(12);
+    tgt_peak_info_buffer = NEW_NOTHROW ObjectBuffer<peak_info>(12);
```

Comment:
If you going to new them in the constructor the might as well be assigned in the header, then they would be guaranteed and you would not have to check at run time.

The allocate would be to allocate them in the first init call so your not paying the memory all the time.

Reply:
@IamPete1 So go back to how I originally had it.  Can you check the current master and tell me if that is what you mean.  I'm not sure what you about allocating them in the first init call

Reply:
@IamPete1 I returned this to original implementation in master.  Please let me know how I would allocate them in the first init call.  I am not familiar with this idea. thanks!